v3/src/components/graph/adornments/univariate-measures/univariate-measure-adornment-component.tsx

```diff
@@ -0,0 +1,245 @@
+      labelObj.label = labelSelection.append("div")
+        .text(`${measureType}=${displayValue}`)
+        .attr("id", labelId)
+        .attr("class", labelClass)
+        .attr("data-testid", labelId)
+        .style("left", `${left}px`)
+        .style("top", `${top}px`)
+
+      labelObj.label.call(
+        drag<HTMLDivElement, unknown>()
+          .on("drag", (e) => handleMoveLabel(e, labelId))
+          .on("end", (e) => handleEndMoveLabel(e, labelId))
+      )
+
+      valueObj.cover.on("mouseover", (e) => highlightLabel(labelId, true))
+        .on("mouseout", (e) => highlightLabel(labelId, false))
+
+      labelObj.label.on("mouseover", (e) => highlightCover(coverId, true))
+        .on("mouseout", (e) => highlightCover(coverId, false))
+    } else {
+      const x = isVertical.current ? xScale(plotValue) + 5 : plotWidth - (plotWidth/2)
+      const y = isVertical.current ? 50 : yScale(plotValue) - 5
+      const textId = `${measureType}-tip-${containerId}${classFromKey ? `-${classFromKey}` : ""}`
+      const textClass = clsx("measure-tip", `${measureType}-tip`)
+      valueObj.text = selection.append("text")
+        .text(`${measureType}=${displayValue}`)
+        .attr("id", textId)
+        .attr("class", textClass)
+        .attr("data-testid", textId)
+        .attr("x", x / xCellCount)
+        .attr("y", y / yCellCount)
+    }
+
+  }, [adornmentsStore?.activeUnivariateMeasures, classFromKey, containerId, handleEndMoveLabel,
+      handleMoveLabel, highlightCover, highlightLabel, layout, measureType, model.type, plotHeight,
+      plotWidth, showLabel, xAttrType, xScale, yAttrType, yScale])
+
+  // Add the lines and their associated covers and labels
+  const refreshValues = useCallback(() => {
+    const disposer = autorun(() => {
+      const { isVisible } = model
+      if (!isVisible) return
+
+      const measure = model?.measures.get(instanceKey)
+      const newValueObj: IValueObject = {}
+      const newLabelObj: ILabelObject = {}
+      const selection = select(valueRef.current)
+      const labelSelection = select(labelRef.current)
+
+      // Remove the previous values' elements
+      selection.html(null)
+      labelSelection.html(null)
+
+      if (measure) {
+        addLineCoverAndLabel(newValueObj, newLabelObj, measure)
+      }
+    }, { name: "UnivariateMeasureAdornmentComponent.refreshValues" })
+    return () => disposer()
+  }, [model, instanceKey, addLineCoverAndLabel])
+
+  // Refresh values on axis changes
+  useEffect(function refreshAxisChange() {
+    return autorun(() => {
+      isVertical.current = dataConfig?.attributeType("x") === "numeric"
+      refreshValues()
+    }, { name: "UnivariateMeasureAdornmentComponent.refreshAxisChange" })
+  }, [dataConfig, refreshValues, xAxis?.max, xAxis?.min, yAxis?.max, yAxis?.min])
+
+  // On unmount, remove measure
+  useEffect(function removeMean() {
+    return () => {
+      model?.removeMeasure(instanceKey)
```

Comment:
Why are we removing measures from the model on unmount? In general, model changes should be triggered by user actions or responding directly to other model changes.

Reply:
I think I was in a mixed up state while trying to troubleshoot a problem when I added this. I've removed it and things work fine.